src/full/Agda/TypeChecking/Serialise.hs

```diff
@@ -44,12 +44,14 @@ import qualified Data.Binary.Get as B
 import qualified Data.Binary.Put as B
 import qualified Data.List as List
 import Data.Function
-import Data.Monoid
+#if !(MIN_VERSION_base(4,11,0))
+import Data.Semigroup((<>))
+#endif
 
 import qualified Codec.Compression.GZip as G
 import qualified Codec.Compression.Zlib.Internal as Z
 
-#if __GLASGOW_HASKELL__ >= 804
+#if __GLASGOW_HASKELL__ >= 821
```

Comment:
Was the bound `804` wrong or why did this get changed?

Reply:
Also, `821` doesn't mean GHC 8.2.1 but GHC 8.21.

Reply:
The bound should be GHC 8.2.1, from my reading of the library dependencies.   But I did indeed read the dependency format incorrectly (`MIN_VERSION_GLASGOW_HASKELL(x,y,z,z')` is the format I was misrememberding.  So 804 is really GHC 8.4.x or higher.  Since it's merely an optimization I'll revert, assuming that someone found that earlier versions were faulty in some way.

Reply:
I removed the `hlint.yaml` file because I found that it was easier the look at the `hlint-report.html` file produced by a full hlint (`make hlint`).  This groups the hints nicely.  (Previously I had switched various hints on and off one by one.)  But it could help to have a default file with the hints that should always be ignored.  For example, we have hundreds of redundant brackets and dollars.  Some people prefer not to hoist nots.  The code base contains a fair fair few ` x : []` instead of `[x]`.  Some seems to people prefer flips to sections. etc.

Reply:
> The bound should be GHC 8.2.1, from my reading of the library dependencies.

There was [some problem](https://github.com/agda/agda/issues/4457#issuecomment-589939292) when I tested with GHC 8.2.2.

> Since it's merely an optimization I'll revert, assuming that someone found that earlier versions were faulty in some way.

I'd like you to consider that four persons have been involved in this discussion, and we don't have unlimited resources.